qiskit/dagcircuit/dagcircuit.py

```diff
@@ -319,6 +319,97 @@ def _add_wire(self, wire):
         else:
             raise DAGCircuitError(f"duplicate wire {wire}")
 
+    def remove_clbits(self, *clbits):
+        """
+        Remove classical bits from the circuit. All bits MUST be idle.
+        Any registers with references to at least one of the specified bits will
+        also be removed.
+
+        Raises:
+            DAGCircuitError: a clbit is not a Clbit, is not in the circuit,
+            or is not idle.
+
+        Returns:
+            set(Clbit): The set of Clbits that were removed.
+        """
+        if any(not isinstance(clbit, Clbit) for clbit in clbits):
+            raise DAGCircuitError("not a Clbit instance.")
+
```

Comment:
I noticed that in other places in this file, we don't seem to attempt to print the object out if we don't know what it is. I wasn't sure if this was a best practice in Python to avoid a potentially very noisy ``__str__`` or ``__repr__``. What would you recommend here? Should I just print it out as normal?

Reply:
Personally I'd usually just do
```python
raise DAGCircuitError(f"not a Clbit instance: '{clbit}'")
```
which uses `str(clbit)`, but it's really up to you.  These messages _should_ only be debugging, so we don't really care _too_ much if passing some crazy object gives you a crazy error message - it still lets you know what sort of object you accidentally passed, which helps narrow down where it might have come in.

It's not a big deal to have it or not have it though.  I just tend towards "if we already have the information, it may help to show it", though I'm certainly not 100% consistent in that.

Reply:
How about something like this?

```py
if any(not isinstance(clbit, Clbit) for clbit in clbits):
    wrong_type_bits = (b for b in clbits if not isinstance(b, Clbit))
    raise DAGCircuitError("clbits not of type Clbit: %s" % wrong_type_bits)
```

Reply:
Yeah, that definitely works!

Reply:
Done in a417d73.